diff -U 8 -pr a/patches/patchinstall.sh b/patches/patchinstall.sh
--- a/patches/patchinstall.sh	2018-07-30 10:09:31.000000000 +0200
+++ b/patches/patchinstall.sh	2018-08-05 11:32:52.466918627 +0200
@@ -245,17 +245,16 @@ patch_enable_all ()
 	enable_ntdll_SystemInterruptInformation="$1"
 	enable_ntdll_SystemRoot_Symlink="$1"
 	enable_ntdll_ThreadTime="$1"
 	enable_ntdll_Threading="$1"
 	enable_ntdll_User_Shared_Data="$1"
 	enable_ntdll_WRITECOPY="$1"
 	enable_ntdll_Wait_User_APC="$1"
 	enable_ntdll_Zero_mod_name="$1"
-	enable_ntdll_futex_condition_var="$1"
 	enable_ntdll_set_full_cpu_context="$1"
 	enable_ntoskrnl_Stubs="$1"
 	enable_nvapi_Stub_DLL="$1"
 	enable_nvcuda_CUDA_Support="$1"
 	enable_nvcuvid_CUDA_Video_Support="$1"
 	enable_nvencodeapi_Video_Encoder="$1"
 	enable_ole32_HGLOBALStream="$1"
 	enable_ole32_STGPROP="$1"
@@ -922,19 +921,16 @@ patch_enable ()
 			enable_ntdll_WRITECOPY="$2"
 			;;
 		ntdll-Wait_User_APC)
 			enable_ntdll_Wait_User_APC="$2"
 			;;
 		ntdll-Zero_mod_name)
 			enable_ntdll_Zero_mod_name="$2"
 			;;
-		ntdll-futex-condition-var)
-			enable_ntdll_futex_condition_var="$2"
-			;;
 		ntdll-set_full_cpu_context)
 			enable_ntdll_set_full_cpu_context="$2"
 			;;
 		ntoskrnl-Stubs)
 			enable_ntoskrnl_Stubs="$2"
 			;;
 		nvapi-Stub_DLL)
 			enable_nvapi_Stub_DLL="$2"
@@ -5449,31 +5445,16 @@ fi
 # |
 if test "$enable_ntdll_Zero_mod_name" -eq 1; then
 	patch_apply ntdll-Zero_mod_name/0001-ntdll-Initialize-mod_name-to-zero.patch
 	(
 		printf '%s\n' '+    { "Qian Hong", "ntdll: Initialize mod_name to zero.", 1 },';
 	) >> "$patchlist"
 fi
 
-# Patchset ntdll-futex-condition-var
-# |
-# | This patchset fixes the following Wine bugs:
-# |   *	[#45524] Add a futex-based implementation of condition variables
-# |
-# | Modified files:
-# |   *	dlls/ntdll/sync.c
-# |
-if test "$enable_ntdll_futex_condition_var" -eq 1; then
-	patch_apply ntdll-futex-condition-var/0001-ntdll-Add-a-futex-based-condition-variable-implement.patch
-	(
-		printf '%s\n' '+    { "Zebediah Figura", "ntdll: Add a futex-based condition variable implementation.", 1 },';
-	) >> "$patchlist"
-fi
-
 # Patchset ntdll-set_full_cpu_context
 # |
 # | Modified files:
 # |   *	dlls/ntdll/signal_i386.c
 # |
 if test "$enable_ntdll_set_full_cpu_context" -eq 1; then
 	patch_apply ntdll-set_full_cpu_context/0001-ntdll-Add-back-SS-segment-prefixes-in-set_full_cpu_c.patch
 	(
