--- a/dlls/wldap32/init.c
+++ b/dlls/wldap32/init.c
@@ -322,7 +322,7 @@ exit:
  *  The timeout parameter may be NULL in which case a default timeout
  *  value will be used.
  */
-ULONG CDECL ldap_connect( WLDAP32_LDAP *ld, struct l_timeval *timeout )
+ULONG CDECL WLDAP32_ldap_connect( WLDAP32_LDAP *ld, struct l_timeval *timeout )
 {
     TRACE( "(%p, %p)\n", ld, timeout );
 
--- a/dlls/wldap32/winldap_private.h
+++ b/dlls/wldap32/winldap_private.h
@@ -320,7 +320,7 @@ ULONG CDECL ldap_compare_ext_sA(WLDAP32_LDAP*,PCHAR,PCHAR,PCHAR,struct WLDAP32_b
 ULONG CDECL ldap_compare_ext_sW(WLDAP32_LDAP*,PWCHAR,PWCHAR,PWCHAR,struct WLDAP32_berval*,PLDAPControlW*,PLDAPControlW*);
 ULONG CDECL ldap_compare_sA(WLDAP32_LDAP*,PCHAR,PCHAR,PCHAR);
 ULONG CDECL ldap_compare_sW(WLDAP32_LDAP*,PWCHAR,PWCHAR,PWCHAR);
-ULONG CDECL ldap_connect(WLDAP32_LDAP*,LDAP_TIMEVAL*);
+ULONG CDECL WLDAP32_ldap_connect(WLDAP32_LDAP*,LDAP_TIMEVAL*);
 WLDAP32_LDAP * CDECL ldap_conn_from_msg(WLDAP32_LDAP*,WLDAP32_LDAPMessage*);
 ULONG CDECL ldap_control_freeA(LDAPControlA*);
 ULONG CDECL ldap_control_freeW(LDAPControlW*);
--- a/dlls/wldap32/wldap32.spec
+++ b/dlls/wldap32/wldap32.spec
@@ -75,7 +75,7 @@
  85 cdecl ldap_compare_ext_sW(ptr wstr wstr wstr ptr ptr ptr)
  86 cdecl ldap_compare_sA(ptr str str str)
  87 cdecl ldap_compare_sW(ptr wstr wstr wstr)
- 88 cdecl ldap_connect(ptr ptr)
+ 88 cdecl ldap_connect(ptr ptr) WLDAP32_ldap_connect
  89 cdecl ldap_control_free(ptr) ldap_control_freeA
  90 cdecl ldap_control_freeA(ptr)
  91 cdecl ldap_control_freeW(ptr)
